src/lib/transition_handler/catchup_scheduler.ml

```diff
@@ -101,19 +121,22 @@ module Make (Inputs : Inputs.S) = struct
       List.(
         filter_opt
         @@ map children ~f:(fun child ->
-               cancel_timeout t (With_hash.hash child) ))
+               cancel_timeout t (With_hash.hash (Cached.peek child)) ))
     in
     List.min_elt remaining_times ~compare:Time.Span.compare
 
-  let watch t ~timeout_duration ~transition =
-    let hash = With_hash.hash transition in
+  let watch t ~timeout_duration ~cached_transition =
+    let hash = With_hash.hash (Cached.peek cached_transition) in
     let parent_hash =
-      With_hash.data transition |> External_transition.Verified.protocol_state
+      With_hash.data (Cached.peek cached_transition)
+      |> External_transition.Verified.protocol_state
       |> Protocol_state.previous_state_hash
     in
     let make_timeout duration =
       Time.Timeout.create t.time_controller duration ~f:(fun _ ->
-          don't_wait_for (Writer.write t.catchup_job_writer transition) )
+          (* it's ok to create a new thread here because the thread essentially does no work *)
```

Comment:
why? Does the `make_timeout` happen more infrequently than the reading from the pipe?

Reply:
I don't think I understand the question. Writing to a pipe is essentially just a block. So ignoring a block basically just means that you aren't binding after it resolves.

Reply:
but if this function is invoked more often than the write resolves we're queueing more async jobs